src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,39 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int TIMEOUT = 5;
 
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
Using CopyOnWriteArrayList is thread-safe, but it can be inefficient for large numbers of listeners due to its write performance characteristics. If writes are frequent, consider using a different concurrent collection or optimizing the concurrency handling.

Reply:
the thing is that listeners are not usually updated too frequently,so I guess it would be fine to use list here, but concurrent map key set can also be a nice fit for our needs, so up to you